Babylon authoritarian home defense and internal security

Unintuition-Time opened the pull request:
Related to #95

Unintuition-Time committed:
Various Initial Steps

added Home Defence lawgroup, moved national guard to law category, made babylon law group and localization

Unintuition-Time committed:
Formatting

Unintuition-Time committed:
added templates for new laws

Unintuition-Time committed:
Added Movements, Law Values

Added values for national guard and localized garrisons

Unintuition-Time committed:
Added Pop support and names

Unintuition-Time committed:
Updated popsupport

Danarca committed:
Split up the law groups for ingame visibility

Unintuition-Time committed:
Localization and Formatting

Unintuition-Time committed:
Localization

Unintuition-Time committed:
Localization and Private Paramilitaries

Unintuition-Time committed:
Internal Security Setup & Initial Laws

also localization

Unintuition-Time committed:
oops

I have been typing the wrong multiplier because i'm dumb

Unintuition-Time committed:
added Provincial Courts + Inquisition

Unintuition-Time committed:
added unlocking laws

Unintuition-Time committed:
added movements

Unintuition-Time committed:
added pop support, progressiveness

Unintuition-Time committed:
Added tech requirements

Unintuition-Time committed:
Reorganization

Unintuition-Time committed:
modified home defence law

Unintuition-Time committed:
Added tech institution levels and localization

Unintuition-Time committed:
Added Citizenship Institution Level

to be carried over and work with the citizenship branch

Unintuition-Time committed:
Failed Localization and small changes

Unintuition-Time committed:
Balance and Localization

Unintuition-Time committed:
Added Decimals and Changed laws

Unintuition-Time commented:
I forgot to do the ideologies, my bad. Please hold review for the next few days while I get that done

Unintuition-Time committed:
Added Ideologies

Added Ideology changes and also modified gendarmerie to be enacted sooner in-line with most law unlocks

Unintuition-Time commented:
Ok now this is ready for review

Unintuition-Time commented:
Shit wait I forgot the new lawgroup's law support :(

Unintuition-Time committed:
Added Ideologies

This section is now considered feature complete, missing only image files, and is ready for review

Unintuition-Time commented:
Finally ready for review

HiddeLekanne commented on common/ideologies/00_ig_ideologies.txt:
```diff
@@ -60,6 +60,15 @@
 		law_true_neutrality = neutral
 	}
+
+	lawgroup_internal_security = {
+		law_provincial_courts = approve
+		law_inquisition = neutral
+		law_no_home_affairs = neutral
+		law_secret_police = disapprove
+		law_empowered_gendarmerie = disapprove
+		law_guaranteed_liberties = strongly_disapprove
+	}	
```
Do you have this in a matrix somehwere? otherwise evaluating this is absolutely a pain.

Unintuition-Time replied:
I do not, I apologize as I was focused on mostly setting everything up properly. I'll work on a matrix this week.

HiddeLekanne commented on common/laws/00_internal_security.txt:
```diff
@@ -1,101 +1,154 @@
+		law_protected_speech
+	}
+	modifier = {
+		#country_institution_home_affairs_max_investment_add = 2
```
No commented out code in pull requests please.

Unintuition-Time replied:
What do you mean by commented out code?

HiddeLekanne commented on common/laws/00_internal_security.txt:
```diff
@@ -1,101 +1,154 @@
+		centralization
+	}
+	unlocking_laws = {
+		law_monarchy
+		law_theocracy
+		law_religious_schools
+		law_state_religion
```
These are OR, so I wouldn't add monarchy in this case, since it would allow for all kinds of weird combinations.

Unintuition-Time replied:
Is there a way to make a "X out of X laws" to meet the requirement system? I asked about it before but never got a response so i went ahead for this. Also, some weird combinations should be possible in my opinion, but if you have an example to put against this I'm happy to receive it.

HiddeLekanne replied:
With monarchy and total separation enacted for example. And no, not way that I know of.

Unintuition-Time replied:
Can we just make total separation and state atheism disallowing laws then?

HiddeLekanne replied:
yes

HiddeLekanne commented on common/laws/00_internal_security.txt:
```diff
@@ -1,101 +1,154 @@
+		law_protected_speech
+	}
+	modifier = {
+		#country_institution_home_affairs_max_investment_add = 2
```
No commented out code in pull requests please.

HiddeLekanne commented on common/laws/00_internal_security.txt:
```diff
@@ -1,101 +1,154 @@
+		country_radicals_from_legitimacy_mult = -0.01
+		country_loyalists_from_legitimacy_mult = 0.05
+		interest_group_ig_devout_pol_str_mult = 0.05
+		country_revolution_progress_mult = 0.1
+		country_secession_progress_mult = 0.1
```
Either the institution promotes or undermines stability, don't make it confusing by doing both. Quicker progression towards a lower radicalism is confused messaging. Especially since an increased progress speed can have weird side effects, like triggering a revolution anyway.

Example:
83 radicalism, current progress 81, progress step = 18 + 10% = 19.2, it will now cause a revolution on the next update because of the 10%.

Unintuition-Time replied:
The Progress Speed is to omit the institution base effect, which grants those. If you load up the game it will show no change as a result.

Since I am merely rescinding a decrease, this effectively does not change the revolution clock at all.

HiddeLekanne replied:
Ah then it's fine.

HiddeLekanne commented on common/laws/00_internal_security.txt:
```diff
@@ -105,39 +158,116 @@ law_national_guard = {
+	}
 	modifier = {
-		country_institution_home_affairs_max_investment_add = 2
+		#country_institution_home_affairs_max_investment_add = 2
```
No commented out code in pull requests please.

HiddeLekanne commented on common/laws/00_internal_security.txt:
```diff
@@ -105,39 +158,116 @@ law_national_guard = {
 		central_archives
 	}
+	unlocking_laws = {
+		law_autocracy
```
oligarchy? single party?

HiddeLekanne commented on common/laws/00_internal_security.txt:
```diff
@@ -105,39 +158,116 @@ law_national_guard = {
+		law_protected_speech
+	}
+	modifier = {
+		#country_institution_home_affairs_max_investment_add = 2
```
No commented out code in pull requests please.

HiddeLekanne commented on common/laws/00_internal_security.txt:
```diff
@@ -105,39 +158,116 @@ law_national_guard = {
+		political_movement_radicalism_mult = -0.05
+		state_radicals_from_sol_change_mult = -0.025
+		state_loyalists_from_sol_change_mult = 0.01
+		state_turmoil_effects_mult = -0.04
```
This is really powerful, I don't know if balanced but this internal security is very good.

Unintuition-Time replied:
What makes it powerful in your opinion?

Unintuition-Time replied:
Made conditions more restrictive to balance

HiddeLekanne commented on common/laws/00_internal_security.txt:
```diff
@@ -105,39 +158,116 @@ law_national_guard = {
+		state_turmoil_effects_mult = -0.04
+		interest_group_ig_armed_forces_pol_str_mult = 0.05
+		interest_group_ig_petty_bourgeoisie_pol_str_mult = 0.02
+		country_revolution_progress_mult = 0.1
+		country_secession_progress_mult = 0.1
```
Again confusing messages by progressing quicker to a lower radicalism.

Unintuition-Time replied:
As stated above, counteracts the institution base modifier.

HiddeLekanne commented on common/laws/00_internal_security.txt:
```diff
@@ -147,45 +277,51 @@ law_secret_police = {
+	}
 	modifier = {
-		country_institution_home_affairs_max_investment_add = 2
+		#country_institution_home_affairs_max_investment_add = 2
```
Please no commented out code

HiddeLekanne reviewed (commented):
Halfway review, now going to sleep.

Unintuition-Time committed:
Feedback Fixes

Removed Commented Out Sections (woops) and added Oligarchy + technocracy + Single Party State to secret Police

Unintuition-Time committed:
Added Armed Forces Home Defence

Forgot this amongst the ideology filling I was doing.

Unintuition-Time committed:
Modified law conditions

better fits the vision for what is and is not allowed to pass these laws.

Unintuition-Time committed:
Nerf to Gendarmerie

Made conditions harder to pass, leading to more restrictive laws that are penalizing to balance the strength of this law

Unintuition-Time committed:
Merge branch 'Babylon-Core' of https://github.com/Babylon-Development-Victoria-3/Victoria-3-Dev into babylon-authoritarian-home-defense-and-internal-security

HiddeLekanne commented on localization/english/babylon_institutions_l_english.yml:
```diff
@@ -0,0 +1,3 @@
+﻿l_english:
+institution_home_defence:0 "Home Defence"
```
missing space

HiddeLekanne commented on common/technology/technologies/babylon_society.txt:
```diff
@@ -70,4 +70,41 @@ scientific_method = {
+}
+
+reactionism = {
+	# New leaders may spawn with the Feminist, Vanguardist Ideology
```
Weird comment?

HiddeLekanne commented on common/technology/technologies/babylon_society.txt:
```diff
@@ -70,4 +70,41 @@ scientific_method = {
+		socialism
+		pan-nationalism
+	}
+	ai_weight = {
+		value = 1
+		if = {
+			limit = {
+				has_strategy = ai_strategy_egalitarian_agenda
+			}
+			add = 1
+		}
+		if = {
+			limit = {
+				OR = {
+					has_strategy = ai_strategy_conservative_agenda
+					has_strategy = ai_strategy_reactionary_agenda
+				}
+			}
+			add = -0.5
+		}
+	}
```
Ai doesn't make sense for reactionism

HiddeLekanne commented on common/modifier_types/babylon_modifier_types.txt:
```diff
@@ -0,0 +1,100 @@
+	num_decimals = 2
+}
+
+state_birth_rate_mult = {
```
Not MILITARY anymore right?

HiddeLekanne commented on common/modifier_types/babylon_modifier_types.txt:
Comments about when you just overwrite the num_decimals or add new modifier_types would be nice.

Unintuition-Time committed:
Pull Request Fixes

Unintuition-Time committed:
AI fix
